crypto/evp/evp_enc.c

```diff
@@ -67,6 +67,8 @@ EVP_CIPHER_CTX *EVP_CIPHER_CTX_new(void)
 
 void EVP_CIPHER_CTX_free(EVP_CIPHER_CTX *ctx)
 {
+    if (ctx == NULL)
+        return;
```

Comment:
Unnecessary, since `EVP_CIPHER_CTX_free() checks for NULL and returns immediately in that case

Reply:
Why we have this check in `EVP_MD_CTX_free()` is a good question, though.  Rather than adding it here, I'd remove it there.

Reply:
I want to have it here so that all free routines follow the same structure.  Since the other function is public, they both need the check.